### ISSUE TITLE
Loosen mixin validation

### ISSUE THREAD

**Issue by jromero** [buildpacks/pack#868]
### Description

Per https://github.com/buildpacks/rfcs/pull/109, the mixin constraints have been loosened. This would apply to `pack build`, `pack create-builder`, `pack rebase` commands.

### Additional context

We may need to make this change Platform API aware and only apply in the Platform API spec this is included in. 

**Blocked by:** 
- [ ] Changes to Platform API spec (https://github.com/buildpacks/spec/issues/149)

**Comment by natalieparellano** [comment]
Closing as no longer relevant, mixins [aren't a thing](https://github.com/buildpacks/rfcs/blob/main/text/0096-remove-stacks-mixins.md) anymore!
